opds2_import.py

```diff
@@ -144,72 +137,63 @@ def _extract_subjects(self, subject):
             )
 
             subject_metadata_list.append(subject_metadata)
-        elif isinstance(subject, list):
-            for inner_subject in subject:
-                inner_subject_metadata_list = self._extract_subjects(inner_subject)
 
-                subject_metadata_list.extend(inner_subject_metadata_list)
+            self._logger.debug(
+                u"Finished extracting subject metadata from {0}: {1}".format(
+                    encode(subject), encode(subject_metadata)
+                )
+            )
 
         self._logger.debug(
-            u"Finished extracting contributor metadata from {0}: {1}".format(
-                encode(subject), encode(subject_metadata_list)
+            u"Finished extracting subjects metadata: {0}".format(
+                encode(subject_metadata_list)
             )
         )
 
         return subject_metadata_list
 
-    def _extract_contributors(self, contributor):
+    def _extract_contributors(self, contributors, default_role=None):
```

Comment:
You don't end up using this, but `contributors.role` can't be null in the database. The default role here should be AUTHOR_ROLE.

Reply:
Changed `default_role`